packages/core/src/routes/interaction/verifications/profile-verification.ts

```diff
@@ -164,63 +169,64 @@ const profileExistValidation = async (
   }
 };
 
-// eslint-disable-next-line complexity
+const isValidRegisterProfile = (profile: Profile): profile is RegisterSafeProfile =>
+  registerProfileSafeGuard.safeParse(profile).success;
+
+const hasPasswordInProfile = (profile: Profile): profile is { password: string } =>
+  Boolean(profile.password);
+
 export default async function profileVerification(
   ctx: InteractionContext,
   provider: Provider,
   interaction: IdentifierVerifiedInteractionResult
-): Promise<IdentifierVerifiedInteractionResult> {
-  if (!interaction.profile && !ctx.interactionPayload.profile) {
-    return interaction;
-  }
-
+): Promise<VerifiedInteractionResult> {
   const profile = { ...interaction.profile, ...ctx.interactionPayload.profile };
 
   const { event, identifiers, accountId } = interaction;
 
-  switch (event) {
-    case Event.Register: {
-      // Verify the profile includes sufficient identifiers to register a new account
-      try {
-        registerProfileSafeGuard.parse(profile);
-      } catch (error: unknown) {
-        throw new RequestError({ code: 'guard.invalid_input' }, error);
-      }
-
-      verifyProfileIdentifiers(profile, identifiers);
-      await profileRegisteredValidation(profile, identifiers);
-      break;
-    }
+  if (event === Event.Register) {
+    // Verify the profile includes sufficient identifiers to register a new account
+    assertThat(isValidRegisterProfile(profile), new RequestError({ code: 'guard.invalid_input' }));
 
-    case Event.SignIn: {
-      verifyProfileIdentifiers(profile, identifiers);
-      // Find existing account
-      const user = await findUserById(accountId);
-      await profileExistValidation(profile, user);
-      await profileRegisteredValidation(profile, identifiers);
-      break;
-    }
+    verifyProfileIdentifiers(profile, identifiers);
+    await profileRegisteredValidation(profile, identifiers);
 
-    case Event.ForgotPassword: {
-      // Forgot Password
-      const { password } = profile;
+    const interactionWithProfile: VerifiedRegisterInteractionResult = { ...interaction, profile };
+    await storeInteractionResult(interactionWithProfile, ctx, provider);
 
-      if (password) {
-        const { passwordEncrypted: oldPasswordEncrypted } = await findUserById(accountId);
+    return interactionWithProfile;
+  }
 
-        assertThat(
-          !oldPasswordEncrypted || !(await argon2Verify({ password, hash: oldPasswordEncrypted })),
-          new RequestError({ code: 'user.same_password', status: 422 })
-        );
-      }
+  if (event === Event.SignIn) {
+    verifyProfileIdentifiers(profile, identifiers);
+    // Find existing account
+    const user = await findUserById(accountId);
+    await profileExistValidation(profile, user);
+    await profileRegisteredValidation(profile, identifiers);
```

Comment:
nit: use `verb-object` expression in function names

Reply:
updated